### PR TITLE
Add Messenger desktop latest

### PR DESCRIPTION
**Pull request by Baltoli**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
    - ~~Gem installs for brew style seem to be broken on Catalina somehow - happy to fix any glaring issues, I just can't manage to get the style checker to run myself.~~
    - The CI style checker reports no issues so I've checked the box
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
    - Previous PR was closed as the author couldn't find the URL - I've got a non-timestamped link for the latest version in this PR.
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/messenger.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  version "73.2.121,249985498"
+  sha256 "104ee1faf4b00d6c77b9500ddd601f6388f25dd781e5dbb5a2c0b73cce68eaa0"
+
+  url "https://www.facebook.com/messenger/desktop/update/#{version.after_comma}.zip"
+  appcast "https://www.facebook.com/messenger/desktop/update/latest-mac.yml"
+  name "Facebook Messenger"
+  desc "Native desktop app for Messenger (formerly Facebook Messenger)"
+  homepage "https://www.messenger.com/desktop"
+
+  app "Messenger.app"
```
Comment (vitorgalvao) [inline Casks/messenger.rb:11]: Please add a [`zap`](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md).

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Review by vitorgalvao — DISMISSED** [review]

**Review by vitorgalvao — DISMISSED** [review]

**Comment by miccal** [comment]
Not sure what is going on with the `appcast` -- in a browser the `.yml` file is downloaded, by via `curl`:
```
|-> curl https://www.facebook.com/messenger/desktop/update/latest-mac.yml
version: 73.2.121
releaseDate: '2020-10-13T06:07:24-07:00'
path: 'https://www.facebook.com/messenger/desktop/update/249985498.zip/'
sha512: 7c8ba308188ff15a03314593bec863ba45c5b7d15f17e4014a1d7d63a54c11a8a3eb2bec0ca36c19ff62b07ed4aa5cda82aef94c41746afff78cd2da70b51c2d
```

**Comment by vitorgalvao** [comment]
`must_contain: version.before_comma` is never needed, because the appcast check always does that (if it’s failing because of that now, there’s a bug that affects a lot more casks.

**Review by vitorgalvao — DISMISSED** [review]

**Review by vitorgalvao — DISMISSED** [review]

**Comment by miccal** [comment]
The `appcast` may not be useful here:
```
|-> curl -sLI https://www.facebook.com/messenger/desktop/update/latest-mac.yml
HTTP/1.1 200 OK
Vary: Accept-Encoding
Pragma: no-cache
Cache-Control: private, no-cache, no-store, must-revalidate
Expires: Sat, 01 Jan 2000 00:00:00 GMT
Strict-Transport-Security: max-age=15552000; preload
Content-Type: application/octet-stream
X-Content-Type-Options: nosniff
X-XSS-Protection: 0
X-Frame-Options: SAMEORIGIN
X-FB-Debug: GSWdL2/66uuv/jOe0jtfZgiNPNR03a5sQ6Sh5eJHA0/i7+6NlqLq6NiKhpTvIIopFlZGhEszEyM3ond2tpfr5w==
Date: Sat, 17 Oct 2020 03:37:37 GMT
Transfer-Encoding: chunked
Alt-Svc: h3-29=":443"; ma=3600,h3-27=":443"; ma=3600
Connection: keep-alive
```
&nbsp;
In particular note the line:
```
Cache-Control: private, no-cache, no-store, must-revalidate
```
&nbsp;
@core-code can you think of a way to utilise this `appcast`?

**Comment by core-code** [comment]
> must_contain: version.before_comma is never needed

for some reason other casks have it too:` bluestacks, get-iplayer-automator qiyimedia qtvlm`

**Comment by core-code** [comment]
> @core-code can you think of a way to utilise this appcast?

sorry, slow thinking today. i don't get the problem. it contains the version number just fine, no?

 ```
~ % curl https://www.facebook.com/messenger/desktop/update/latest-mac.yml
version: 73.2.121
[…]
```

**Review by miccal — APPROVED** [review]

**Comment by miccal** [comment]
Thank you @Baltoli.

**Pull request merged**